portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -58,21 +76,46 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get name and text
-    String name = request.getParameter("name-input");
-    String text = request.getParameter("text-input");
-    if (text != null && name != null) {
-      long timestamp = System.currentTimeMillis();
-
-      Entity messageEntity = new Entity(Constants.MESSAGE_ENTITY_TYPE);
-      messageEntity.setProperty("name", name);
-      messageEntity.setProperty("text", text);
-      messageEntity.setProperty("timestamp", timestamp);
-
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      datastore.put(messageEntity);
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+        String userEmail = userService.getCurrentUser().getEmail();
+        
+        // Get name and text
+        String name = getUserNickname(userService.getCurrentUser().getUserId());
+        String text = request.getParameter("text-input");
+        if (text != null && name != null) {
```

Comment:
1) think about what happens if this is false - what would be a better behavior?
2) indentation

Reply:
To throw an error?

Reply:
Yep! Let's throw an error